### PR TITLE
Add shooting star tear effect

### PR DESCRIPTION
**Pull request by jtmcginnis**
Increase chance of shooting star

### CODE REVIEW

File: main.lua [hunk 1]
```diff
@@ -757,14 +757,23 @@ end
...
-    AlphaAPI.addFlag(tear, ENTITY_FLAGS.SHOOTINGSTAR_TEAR)
-    --Change sprite once it's available
-  end
+    if not AlphaAPI.hasFlag(tear, ENTITY_FLAGS.SHOOTINGSTAR_TEAR) and AlphaAPI.getLuckRNG(35, 4) then
+        AlphaAPI.addFlag(tear, ENTITY_FLAGS.SHOOTINGSTAR_TEAR)
+        local tear_sprite = tear:GetSprite()
+        tear_sprite:Load("gfx/animations/effects/animation_tears_shootingstar.anm2",true)
+        tear_sprite:Play("MoveHori")
+        tear_sprite.Scale = tear.SpriteScale * 0.5
+        tear_sprite:LoadGraphics()
```
Comment (cathery1) [inline main.lua]: you don' t need to load twice you already loaded graphics in sprite:load

File: main.lua [hunk 2]
```diff
@@ -757,14 +757,23 @@ end
...
+    end
 end
 
 function Alphabirth.shootingStarTearUpdate(tear, data)
+    if AlphaAPI.hasFlag(tear, ENTITY_FLAGS.SHOOTINGSTAR_TEAR) then
+        tear.SpriteRotation = tear.Velocity:GetAngleDegrees()
```
Comment (cathery1) [inline main.lua]: i think you only need to set the sprite rotation in tear spawn

File: main.lua [hunk 3]
```diff
@@ -757,14 +757,23 @@ end
...
-  if not AlphaAPI.hasFlag(tear, ENTITY_FLAGS.SHOOTINGSTAR_TEAR) and AlphaAPI.getLuckRNG(15, 4) then
-    AlphaAPI.addFlag(tear, ENTITY_FLAGS.SHOOTINGSTAR_TEAR)
-    --Change sprite once it's available
-  end
+    if not AlphaAPI.hasFlag(tear, ENTITY_FLAGS.SHOOTINGSTAR_TEAR) and AlphaAPI.getLuckRNG(35, 4) then
+        AlphaAPI.addFlag(tear, ENTITY_FLAGS.SHOOTINGSTAR_TEAR)
+        local tear_sprite = tear:GetSprite()
+        tear_sprite:Load("gfx/animations/effects/animation_tears_shootingstar.anm2",true)
+        tear_sprite:Play("MoveHori")
+        tear_sprite.Scale = tear.SpriteScale * 0.5
```
Comment (cathery1) [inline main.lua]: bad idea

Comment (jtmcginnis) [inline main.lua]: still do not know what you're referring to. The scaling?

File: main.lua (hunk 1 above)
Comment (jtmcginnis) [inline main.lua]: that's a good catch

File: main.lua (hunk 2 above)
Comment (jtmcginnis) [inline main.lua]: Setting it only on spawn does not work

**Pull request merged**